store/localcachelayer/team_layer.go

```diff
@@ -70,6 +70,7 @@ func (s LocalCacheTeamStore) Update(team *model.Team) (*model.Team, *model.AppEr
 	if err != nil {
 		return nil, err
 	}
+	defer s.rootStore.doClearCacheCluster(s.rootStore.rolePermissionsCache)
```

Comment:
@mkraft, I'm sure this works and it makes sense, however I'm not clear what are the perf implications to this- if any  (clearing the permissions cache on every team update, considering that we don't seem to be doing it elsewhere on Update - like channels). Also, not clear when we do InvalidateCache versus ClearCache (e.g currently, when we delete a role we invalidate the role cache but clear the permissions cache) and when we defer and when we don't, but I'm new to this area.

Reply:
@catalintomai The `rolePermissionsCache` is caching a small number of keys, and actually more than necessary (for code simplicity). It's also invalidating more than is necessary (again for simplicity). At this stage in the beta release of channel moderation I am not too concerned with the extra cache memory being used or the performance impact of a few extra DB calls when the `Teams` records are updated—something that doesn't happen _too_ frequent. My main concern is it functioning properly and then in future iterations tweaking performance.

I am also new to the `localcachelayer` package, but my understanding is that `(*LocalCacheStore).doInvalidateCacheCluster` is for invalidating a single key and `(*LocalCacheStore).doClearCacheCluster` is for invalidating an entire cache.